src/saml.ts

```diff
@@ -1226,12 +1226,12 @@ class SAML {
         if (!restriction.Audience || !restriction.Audience[0] || !restriction.Audience[0]._) {
           return new Error("SAML assertion AudienceRestriction has no Audience value");
         }
-        if (restriction.Audience[0]._ !== expectedAudience) {
+        if (!restriction.Audience.some(audience => audience._ === expectedAudience)) {
           return new Error(
             "SAML assertion audience mismatch. Expected: " +
               expectedAudience +
               " Received: " +
-              restriction.Audience[0]._,
+              restriction.Audience.map(_ => _._).join(", "),
```

Comment:
This seems crazy, but could we change this so that we don't use `_` as an argument to a function that is used. Typically `_` as an argument would indicate that it is throw-away or unused.

```suggestion
              restriction.Audience.map(a => a._).join(", "),
```